server/chat-plugins/chatlog.ts

```diff
@@ -429,10 +427,23 @@ export const LogViewer = new class {
 	}
 };
 
-/** match with two lines of context in either direction */
+/** Match with two lines of context in either direction */
 type SearchMatch = readonly [string, string, string, string, string];
 
-const LogSearcher = new class {
+export const LogSearcher = new class {
+	constructRegex(str: string) {
+		const searches = str.split('+');
+		if (str.length <= 3 && searches.length < 1) return `\b${str}`;
+		if (searches.length > 1) {
+			let regex = '^';
+			for (const term of searches) {
+				regex += `(?=.*${term})`;
+			}
+			return regex;
+		}
+		return str;
+	}
```

Comment:
```suggestion
	constructRegex(str: string) {
		const searches = str.split('+');
		if (searches.length <= 1) {
			if (str.length <= 3) return `\b${str}`;
			return str;
		}

		return `^` + searches.map(term => `(?=.*${term})`).join('');
	}
```

`searches.length < 1` will always be false; `''.split('+')` is still `['']`, not `[]`. You meant `searches.length <= 1`. The funny thing is that Annika didn't notice, either. :p